File: easy/0349. Intersection of Two Arrays/main.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* intersection(int* nums1, int nums1Size, int* nums2, int nums2Size, int* returnSize) {
    int* result = (int*)malloc(sizeof(int) * (nums1Size < nums2Size ? nums1Size : nums2Size));

    int count = 0;
    for (int i = 0; i < nums1Size; i++) {
        for (int j = 0; j < nums2Size; j++) {
            if (nums1[i] == nums2[j]) {
                bool duplicate = false;
                for (int k = 0; k < count; k++) {
                    if (result[k] == nums1[i]) {
                        duplicate = true;
                        break;
                    }
                }

                if (!duplicate) {
                    result[count++] = nums1[i];
                }

                break;
            }
        }
    }

    *returnSize = count;

    return result;
}

File: easy/0349. Intersection of Two Arrays/main.rs
use std::collections::HashSet;

impl Solution {
    pub fn intersection(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<i32> {
        let mut s1 = HashSet::new();
        let mut s2 = HashSet::new();

        for val in nums1 {
            s1.insert(val);
        }
        for val in nums2 {
            s2.insert(val);
        }

        let mut result: Vec<i32> = Vec::new();
        for val in s1 {
            if s2.contains(&val) {
                result.push(val);
            }
        }

        result
    }
}
